tests/platform_tests/test_engine.py

```diff
@@ -254,11 +254,11 @@ def setUp(self):
         # when switching context. We'll use them layer to count how many times
         # they have been invoked and with what parameters.
         self._pre_patch = mock.patch(
-            "sgtk.platform.engine._execute_pre_context_change_hook",
```

Comment:
So, I've been having this weird issue with mocked code and using tank vs sgtk. After adding the code above and the unit test, all sorts of tests starting breaking when unit tests were run as a whole, but not when they were being run individually. I think the mock module has issues restoring the original methods when mocking something sgtk.

So you'll changes to some patches that get us out of this weird behaviour.